### PR TITLE
Ephemeris in plugin

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: ksp_plugin/celestial_body.hpp [hunk 1]
```diff
@@ -5,114 +5,65 @@
...
-  celestial->history_ =
-      Trajectory<Barycentric>::ReadFromMessage(
-          message.history_and_prolongation().history(),
-          celestial->body_.get());
-  celestial->prolongation_ =
-      Trajectory<Barycentric>::ReadPointerFromMessage(
-          message.history_and_prolongation().prolongation(),
-          celestial->history_.get());
-  return celestial;
+inline void Celestial::set_parent(not_null<Celestial const*> const parent) {
```
Comment (pleroy) [inline ksp_plugin/celestial_body.hpp]: I am wondering: how hard would it be to set the parent at construction?

File: ksp_plugin/interface.cpp [hunk 2]
```diff
@@ -333,9 +333,14 @@ void principia__set_prediction_length(Plugin* const plugin,
   CHECK_NOTNULL(plugin)->set_prediction_length(t * Second);
 }
 
-void principia__set_prediction_step(Plugin* const plugin,
-                                    double const t) {
-  CHECK_NOTNULL(plugin)->set_prediction_step(t * Second);
+void principia__set_prediction_length_tolerance(Plugin* const plugin,
+                                      double const t) {
```
Comment (pleroy) [inline ksp_plugin/interface.cpp]: Bad indent.

File: ksp_plugin/interface.cpp [hunk 3]
```diff
@@ -333,9 +333,14 @@ void principia__set_prediction_length(Plugin* const plugin,
...
+  CHECK_NOTNULL(plugin)->set_prediction_length_tolerance(t * Metre);
+}
+
+void principia__set_prediction_speed_tolerance(Plugin* const plugin,
+                                      double const t) {
```
Comment (pleroy) [inline ksp_plugin/interface.cpp]: Same.

File: ksp_plugin/plugin.hpp [hunk 4]
```diff
@@ -31,15 +32,18 @@ using geometry::Displacement;
 using geometry::Instant;
 using geometry::Point;
 using geometry::Rotation;
-using integrators::SPRKIntegrator;
+using integrators::FixedStepSizeIntegrator;
+using integrators::AdaptiveStepSizeIntegrator;
 using physics::Body;
 using physics::FrameField;
-using physics::NBodySystem;
+using physics::Ephemeris;
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Alphabetize.

File: ksp_plugin/plugin.hpp [hunk 5]
```diff
@@ -365,8 +379,9 @@ class Plugin {
       not_null<RenderingTransforms*>const transforms,
       Position<World> const& sun_world_position) const;
 
-  // TODO(egg): Constant time step for now.
   Time const Δt_ = 10 * Second;
+  Length const prolongation_length_tolerance = 1 * Milli(Metre);
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Missing_ trailing_ underscore_ (2x)

File: ksp_plugin/plugin.hpp [hunk 6]
```diff
@@ -387,15 +402,26 @@ class Plugin {
   // Only one prediction for now, using constant timestep.
   Vessel* predicted_vessel_ = nullptr;
   Time prediction_length_ = 1 * Hour;
-  Time prediction_step_ = Δt_;
+  Length prediction_length_tolerance = 1 * Metre;
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Same_

File: ksp_plugin/plugin.hpp [hunk 7]
```diff
@@ -403,8 +429,13 @@ class Plugin {
...
   // The current in-game universal time.
   Instant current_time_;
+  // The common last time of the histories of synchronized vessels.
+  // TODO(egg): test the serialization of that guy, found out that it wasn't
+  // serialized thanks to vessel invariant violation.  Perhaps check that
+  // a deserialized plugin still functions normally.
+  Instant history_time_;
 
-  not_null<Celestial*> const sun_;  // Not owning.
+  Celestial* sun_;  // Not owning, not null after construction.
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: I am sad that an invariant gets weakened.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 8]
```diff
@@ -726,13 +728,7 @@ where is_in_physics_bubble(vessel)
...
     rendered_prediction_ = new VectorLine(
         lineName     : "rendered_prediction_",
-        linePoints   : new UnityEngine.Vector3[
-                           Math.Min(
-                               kMaxVectorLinePoints,
-                               2 * (int)(prediction_lengths_[
-                                             prediction_length_index_] /
-                                         prediction_step_sizes_[
-                                             prediction_step_index_]))],
+        linePoints   : new UnityEngine.Vector3[1000],  // TODO
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Put a username and a decent text for the TODO.

File: ksp_plugin_test/celestial_test.cpp [hunk 9]
```diff
@@ -15,51 +15,37 @@ namespace ksp_plugin {
...
-}
-
-TEST_F(CelestialDeathTest, SerializationError) {
+TEST_F(CelestialDeathTest, OverlyInitialized) {
   EXPECT_DEATH({
-    serialization::Celestial message;
-    celestial_->WriteToMessage(&message);
-  }, "is_initialized");
+      celestial_->set_trajectory(&trajectory_);
+      celestial_->set_trajectory(&trajectory_);}, "!is_initialized");
```
Comment (pleroy) [inline ksp_plugin_test/celestial_test.cpp]: New line before }, string on its own line.

File: ksp_plugin_test/interface_test.cpp [hunk 10]
```diff
@@ -472,8 +472,6 @@ TEST_F(InterfaceTest, PredictionGettersAndSetters) {
   principia__clear_predicted_vessel(plugin_.get());
   EXPECT_CALL(*plugin_, set_prediction_length(42 * Second));
   principia__set_prediction_length(plugin_.get(), 42);
-  EXPECT_CALL(*plugin_, set_prediction_step(20 * Milli(Second)));
```
Comment (pleroy) [inline ksp_plugin_test/interface_test.cpp]: Y U NO test the two new setters?

File: ksp_plugin_test/physics_bubble_test.cpp [hunk 11]
```diff
@@ -1,4 +1,4 @@
-#include "ksp_plugin/physics_bubble.hpp"
+﻿#include "ksp_plugin/physics_bubble.hpp"
```
Comment (pleroy) [inline ksp_plugin_test/physics_bubble_test.cpp]: No idea what changed here...

File: ksp_plugin_test/physics_bubble_test.cpp [hunk 12]
```diff
@@ -181,11 +190,25 @@ class PhysicsBubbleTest : public testing::Test {
...
+                                           191 * SIUnit<Length>(),
+                                           193 * SIUnit<Length>()}),
+                      8));;
+    } else if (mutable_trajectory->last().time() == t2_) {
+      EXPECT_THAT(bubble_.DisplacementCorrection(
+                      rotation_, celestial_, celestial_world_position_),
+                  AlmostEquals(
+                      Displacement<World>({-27.5 * SIUnit<Length>(),
+                                           193 * SIUnit<Length>(),
+                                           196 * SIUnit<Length>()}),
```
Comment (pleroy) [inline ksp_plugin_test/physics_bubble_test.cpp]: I suppose that these numbers were validated with Mathematica or some such?

File: physics/ephemeris.hpp [hunk 13]
```diff
@@ -32,6 +32,7 @@ class Ephemeris {
   // The equation describing the motion of the |bodies_|.
   using NewtonianMotionEquation =
       SpecialSecondOrderDifferentialEquation<Position<Frame>>;
+  using Trajectories = std::vector<not_null<Trajectory<Frame>*>>;
```
Comment (pleroy) [inline physics/ephemeris.hpp]: I am not too happy with this change, as it seems rather random to put this here since Ephemeris doesn't really deal with Trajectory objects.  When I saw a reference in Plugin I had to look to see if this was a continuous or a discrete trajectory.  I suggest moving the typedef to plugin.h.

File: physics/ephemeris.hpp [hunk 14]
```diff
@@ -43,59 +44,68 @@ class Ephemeris {
...
-      Instant const& t);
+  virtual void FlowWithFixedStep(Trajectories const& trajectories,
+                                 Time const& step,
+                                 Instant const& t);
 
-  void WriteToMessage(
+  virtual void WriteToMessage(
       not_null<serialization::Ephemeris*> const message) const;
-  static not_null<std::unique_ptr<Ephemeris>> ReadFromMessage(
+  static std::unique_ptr<Ephemeris> ReadFromMessage(
```
Comment (pleroy) [inline physics/ephemeris.hpp]: Why the change?  If it can return a nullptr it would be good to document why/how/when.

File: physics/ephemeris_body.hpp [hunk 15]
```diff
@@ -29,7 +29,7 @@ using ::std::placeholders::_3;
 
 namespace physics {
 
-namespace {
+namespace {  // TODO(egg): this should be a named namespace (internal)
```
Comment (pleroy) [inline physics/ephemeris_body.hpp]: Could you create an issue instead of the TODO?  This is not the only place where we have this situation and I remain unconvinced that a namespace internal would help much.  (I'd prefer a namespace named after the file name.)

File: physics/ephemeris_body.hpp [hunk 16]
```diff
@@ -315,7 +342,7 @@ not_null<std::unique_ptr<Ephemeris<Frame>>> Ephemeris<Frame>::ReadFromMessage(
       bodies.size(),
       DegreesOfFreedom<Frame>(Position<Frame>(), Velocity<Frame>()));
   Instant const initial_time;
-  not_null<std::unique_ptr<Ephemeris<Frame>>> ephemeris =
```
Comment (pleroy) [inline physics/ephemeris_body.hpp]: Could we keep the not_null by writing ephemeris(...)?

File: physics/mock_ephemeris.hpp [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+namespace physics {
+
+template<typename Frame>
+class MockEphemeris : public Ephemeris<Frame> {
+ public:
+  MockEphemeris() : Ephemeris() {}
+
+  MOCK_CONST_METHOD0_T(bodies, std::vector<MassiveBody const*> const&());
+  MOCK_CONST_METHOD1_T(
+      trajectory,
```
Comment (pleroy) [inline physics/mock_ephemeris.hpp]: Move to the previous line and reindent the 2nd parameter, it seems that it would fit.

File: serialization/ksp_plugin.proto [hunk 18]
```diff
@@ -60,11 +61,20 @@ message Plugin {
     required Celestial celestial = 2;
     optional int32 parent_index = 3;
   }
+  message CelestialParenthood {
+    required int32 index = 1;
+    optional  int32 parent_index = 2;
```
Comment (pleroy) [inline serialization/ksp_plugin.proto]: Extra space.

File: physics/ephemeris_body.hpp (hunk 16 above)
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: NO.

File: physics/ephemeris_body.hpp (hunk 15 above)
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: Keeping the TODO, making an issue.

File: ksp_plugin_test/physics_bubble_test.cpp (hunk 11 above)
Comment (eggrobin) [inline ksp_plugin_test/physics_bubble_test.cpp]: BOM probably

File: serialization/ksp_plugin.proto (hunk 18 above)
Comment (eggrobin) [inline serialization/ksp_plugin.proto]: Done.

File: physics/mock_ephemeris.hpp (hunk 17 above)
Comment (eggrobin) [inline physics/mock_ephemeris.hpp]: Done.

File: physics/ephemeris.hpp (hunk 14 above)
Comment (eggrobin) [inline physics/ephemeris.hpp]: It cannot be null. Documented.

File: physics/ephemeris.hpp (hunk 13 above)
Comment (eggrobin) [inline physics/ephemeris.hpp]: Done.

File: ksp_plugin_test/physics_bubble_test.cpp (hunk 12 above)
Comment (eggrobin) [inline ksp_plugin_test/physics_bubble_test.cpp]: They make sense.

File: ksp_plugin_test/interface_test.cpp (hunk 10 above)
Comment (eggrobin) [inline ksp_plugin_test/interface_test.cpp]: Done.

File: ksp_plugin_test/celestial_test.cpp (hunk 9 above)
Comment (eggrobin) [inline ksp_plugin_test/celestial_test.cpp]: Moo.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 8 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin/plugin.hpp (hunk 7 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: So am I.

File: ksp_plugin/plugin.hpp (hunk 5 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 6 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Edno.

File: ksp_plugin/interface.cpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: Done.

File: ksp_plugin/interface.cpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: Done.

File: ksp_plugin/celestial_body.hpp (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin/celestial_body.hpp]: We could, but we still need a setter, unless UpdateCelestialHierarchy disappears. I'm not sure whether we need that one, but let's not look at this now.

File: ksp_plugin/plugin.cpp [hunk 19]
```diff
@@ -89,14 +101,37 @@ void Plugin::InsertCelestial(
...
+      FindOrDie(*initial_state_, parent_index);
+  initial_state_->emplace(celestial_index,
+                          parent_degrees_of_freedom + relative);
 }
 
 void Plugin::EndInitialization() {
   initializing_.Flop();
+  std::vector<not_null<std::unique_ptr<MassiveBody const>>> bodies;
+  std::vector<DegreesOfFreedom<Barycentric>> initial_state;
+  for (auto& body : *bodies_) {
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Call the loop variable pair and give pair.second the name body.

File: ksp_plugin/plugin.cpp [hunk 20]
```diff
@@ -89,14 +101,37 @@ void Plugin::InsertCelestial(
...
+  bodies_.reset();
+  for (auto const& state : *initial_state_) {
+    initial_state.emplace_back(state.second);
+  }
+  initial_state_.reset();
+  ephemeris_ = std::make_unique<Ephemeris<Barycentric>>(std::move(bodies),
+                                                        initial_state,
+                                                        current_time_,
+                                                        history_integrator_,
+                                                        45 * Minute,
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Comment the last two parameters.

File: ksp_plugin/plugin.cpp [hunk 21]
```diff
@@ -89,14 +101,37 @@ void Plugin::InsertCelestial(
...
+                                                        1 * Milli(Metre));
+  for (auto const& index_celestial : celestials_) {
+    auto& celestial = *index_celestial.second;
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Preference for naming the type here.

File: ksp_plugin/plugin.cpp [hunk 22]
```diff
@@ -89,14 +101,37 @@ void Plugin::InsertCelestial(
...
+    // TODO(egg): unorthodox address of reference.
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: https://www.youtube.com/watch?v=TGPYBb2DRsg

File: ksp_plugin/plugin.cpp [hunk 23]
```diff
@@ -89,14 +101,37 @@ void Plugin::InsertCelestial(
...
+    initial_state.emplace_back(state.second);
+  }
+  initial_state_.reset();
+  ephemeris_ = std::make_unique<Ephemeris<Barycentric>>(std::move(bodies),
+                                                        initial_state,
+                                                        current_time_,
+                                                        history_integrator_,
+                                                        45 * Minute,
+                                                        1 * Milli(Metre));
+  for (auto const& index_celestial : celestials_) {
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: In general, preference for pair as the loop variable, it makes it clear that we iterate over a map.

File: ksp_plugin/plugin.cpp [hunk 24]
```diff
@@ -294,30 +343,30 @@ bool Plugin::has_vessel(GUID const& vessel_guid) const {
 not_null<std::unique_ptr<RenderingTransforms>>
 Plugin::NewBodyCentredNonRotatingTransforms(
     Index const reference_body_index) const {
+  CHECK(!initializing_);
   // TODO(egg): this should be const, use a custom comparator in the map.
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: This comment is probably obsolete.

File: ksp_plugin/plugin.cpp [hunk 25]
```diff
@@ -416,7 +467,6 @@ void Plugin::WriteToMessage(
     not_null<Celestial const*> const celestial = index_celestial.second.get();
     auto const celestial_message = message->add_celestial();
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: auto*

File: ksp_plugin/plugin.cpp [hunk 26]
```diff
@@ -416,7 +467,6 @@ void Plugin::WriteToMessage(
...
     celestial_message->set_index(index);
-    celestial->WriteToMessage(celestial_message->mutable_celestial());
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: This change confuses me.

File: ksp_plugin/plugin.cpp [hunk 27]
```diff
@@ -444,19 +500,28 @@ void Plugin::WriteToMessage(
...
     serialization::Plugin const& message) {
   LOG(INFO) << __FUNCTION__;
+  auto ephemeris = Ephemeris<Barycentric>::ReadFromMessage(message.ephemeris());
+  auto const& bodies = ephemeris->bodies();
+  auto bodies_it = bodies.begin();
   IndexToOwnedCelestial celestials;
   for (auto const& celestial_message : message.celestial()) {
-    celestials.emplace(
-        celestial_message.index(),
-        Celestial::ReadFromMessage(celestial_message.celestial()));
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: ... as does this one.

File: ksp_plugin/plugin.cpp [hunk 28]
```diff
@@ -444,19 +500,28 @@ void Plugin::WriteToMessage(
...
+    auto const inserted =
+        celestials.emplace(celestial_message.index(),
+                           make_not_null_unique<Celestial>(*bodies_it));
+    CHECK(inserted.second);
+    inserted.first->second->set_trajectory(ephemeris->trajectory(*bodies_it));
+    ++bodies_it;
   }
+  CHECK_EQ(bodies.end() - bodies.begin(), bodies_it - bodies.begin());
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Why not just bodies_it == bodies.end()?

File: ksp_plugin/plugin.cpp [hunk 29]
```diff
@@ -649,28 +708,25 @@ void Plugin::EvolveHistories(Instant const& t) {
...
-                            t,                     // tmax
-                            Δt_,                   // Δt
-                            0,                     // sampling_period
-                            false,                 // tmax_is_exact
-                            trajectories);         // trajectories
-  CHECK_GE(HistoryTime(), current_time_);
+          << "from : " << history_time_;
+  ephemeris_->FlowWithFixedStep(histories, Δt_, t);
+  history_time_ = histories.front()->last().time();
+  CHECK_GE(history_time_, current_time_);
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Should there be a check comparing t and history_time_?

File: ksp_plugin/plugin.cpp [hunk 30]
```diff
@@ -739,24 +796,16 @@ void Plugin::ResetProlongations() {
...
   VLOG(1) << "Prolongations have been reset";
 }
 
 void Plugin::EvolveProlongationsAndBubble(Instant const& t) {
   VLOG(1) << __FUNCTION__ << '\n' << NAMED(t);
-  NBodySystem<Barycentric>::Trajectories trajectories;
-  trajectories.reserve(vessels_.size() + celestials_.size() -
+  Trajectories trajectories;
+  trajectories.reserve(vessels_.size() -
                        bubble_->number_of_vessels() + bubble_->size());
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Maybe give a name (intermediate constant) to size - number_of_vessels.

File: ksp_plugin_test/plugin_test.cpp [hunk 31]
```diff
@@ -91,32 +90,59 @@ MATCHER_P(HasNonvanishingIntrinsicAccelerationAt, t, "") {
...
       : Plugin(initial_time,
                sun_index,
                sun_gravitational_parameter,
-               planetarium_rotation) {
-    n_body_system_.reset(n_body_system);
+               planetarium_rotation),
+        mock_ephemeris_(
+            std::make_unique<StrictMock<MockEphemeris<Barycentric>>>()) {}
+
+  void EndInitialization() override {
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: Can I haz a comment for this method?

File: ksp_plugin_test/plugin_test.cpp [hunk 32]
```diff
@@ -544,48 +568,67 @@ TEST_F(PluginTest, AdvanceTimeWithVessels) {
...
+              Prolong(HistoryTime(sync_time, 0) + δt))
+      .RetiresOnSaturation();
+  // Called to compute the prolongation.
+  EXPECT_CALL(*mock_ephemeris_,
+              FlowWithAdaptiveStep(_, _, _, _, HistoryTime(sync_time, 0) + δt))
+      .RetiresOnSaturation();
+  // Therer are no old vessels, this will make all the new vessels old
+  // instantly without performing any syncing computations.
+  plugin_->AdvanceTime(HistoryTime(sync_time, 0) + δt,
+                       planetarium_rotation);
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: Would fit on the previous line.

File: ksp_plugin_test/plugin_test.cpp [hunk 33]
```diff
@@ -715,20 +765,42 @@ TEST_F(PluginTest, PhysicsBubble) {
...
+  Instant sync_time = initial_time_;
+  // The synchronizer vessel.
+  InsertVessel(constantinople,
+               &expected_number_of_clean_old_vessels,
+               HistoryTime(sync_time, 0));
+  // Called to compute the prolongation.
+  EXPECT_CALL(*mock_ephemeris_,
+              FlowWithAdaptiveStep(_, _, _, _, HistoryTime(sync_time, 0) + δt))
+      .RetiresOnSaturation();
+  // Therer are no old vessels, this will make all the new vessels old
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: There

File: ksp_plugin_test/plugin_test.cpp [hunk 34]
```diff
@@ -1108,13 +1174,12 @@ TEST_F(PluginTest, BarycentricRotatingRenderingIntegration) {
 }
 
 // Checks that we correctly predict a full circular orbit around a massive body
-// with unit gravitational parameter at unit distance, in 8 steps.  Since
+// with unit gravitational parameter at unit distance.  Since
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: Move some words at the end of this line.

Comment (eggrobin) [inline ksp_plugin_test/plugin_test.cpp]: Doen.

File: ksp_plugin_test/plugin_test.cpp (hunk 33 above)
Comment (eggrobin) [inline ksp_plugin_test/plugin_test.cpp]: "ll should be the comment syntax"

File: ksp_plugin_test/plugin_test.cpp (hunk 32 above)
Comment (eggrobin) [inline ksp_plugin_test/plugin_test.cpp]: Done.

File: ksp_plugin_test/plugin_test.cpp (hunk 31 above)
Comment (eggrobin) [inline ksp_plugin_test/plugin_test.cpp]: Yes.

File: ksp_plugin/plugin.cpp (hunk 30 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done. Also `count`, `size` was silly.

File: ksp_plugin/plugin.cpp (hunk 29 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: No.

File: ksp_plugin/plugin.cpp (hunk 28 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Better output this way,.

File: ksp_plugin/plugin.cpp (hunk 27 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Yes, but that sort of cleanup will get done when we rewrite the plugin for Brouwer.

File: ksp_plugin/plugin.cpp (hunk 25 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Doen.

File: ksp_plugin/plugin.cpp (hunk 24 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: WTF

File: ksp_plugin/plugin.cpp (hunk 21 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: We don't do that when iterating over vectors, so no.

File: ksp_plugin/plugin.cpp (hunk 23 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Ok. Eventually we'll have to do the same elsewhere.

File: ksp_plugin/plugin.cpp (hunk 20 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 19 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

**Comment by eggrobin** [comment]
This fixes #228. We'll want to unfix it.
